buildSrc/src/main/kotlin/uk/gov/justice/digital/hmpps/plugins/JibConfigPlugin.kt

```diff
@@ -34,23 +38,33 @@ class JibConfigPlugin : Plugin<Project> {
                             includes.add("agent/agent.jar")
                         }
                         path {
-                            setFrom("${project.projectDir}")
+                            setFrom("${project.buildDir}/agent")
                             includes.add("applicationinsights*.json")
                             into = "/agent"
                         }
                     }
                 }
             }
 
-            val copyAgentTask = project.rootProject.tasks.named("copyAgent")
-            project.tasks.named("jib") {
-                dependsOn(copyAgentTask)
+            val copyAgent = project.rootProject.tasks.named("copyAgent")
+            val copyAppInsightsConfig = project.tasks.register<Copy>("copyAppInsightsConfig") {
+                from("${project.projectDir}/applicationinsights.json")
+                into("${project.buildDir}/agent")
+                inputs.file("${project.projectDir}/applicationinsights.json")
+                outputs.file("${project.buildDir}/agent/applicationinsights.json")
+                outputs.cacheIf { true }
             }
-            project.tasks.named("jibBuildTar") {
-                dependsOn(copyAgentTask)
-            }
-            project.tasks.named("jibDockerBuild") {
-                dependsOn(copyAgentTask)
+            val assemble = project.tasks.named("assemble")
+            project.tasks.withType<BuildImageTask>().named("jib") {
+                doFirst {
+                    jib!!.to.tags = setOf("${project.version}")
+                }
+                dependsOn(copyAgent, `copyAppInsightsConfig`, assemble)
+                inputs.dir("deploy")
+                inputs.dir("src")
+                inputs.file("build.gradle.kts")
```

Comment:
Could we pick this up again today?  It occurred to me that this won't build if any of the libs or dependency versions change.